packages/material-ui/src/FormControl/FormControl.test.js

```diff
@@ -164,26 +178,6 @@ describe('<FormControl />', () => {
       );
     });
 
-    describe('from state', () => {
```

Comment:
State isn't a public API I don't think we should test it

Reply:
I guess the better approach would be to construct a component that interacts with the context how "it's supposed to" and then make assertions about that component. 

Since we don't expose the context anyway a good integration test in `TextField` should create the best test with regards to confidence. Essentially a unit test is useless if the component is only used with other components from our library. There's no unit here but only integration.

But that's a little bit out of scope. Maybe look at the coverage report and make a judgement call if it's good enough.